tests/std/tests/P0896R4_ranges_alg_fill/test.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <range_algorithm_support.hpp>
+#include <ranges>
+#include <utility>
+
+constexpr void smoke_test() {
+    using ranges::fill, ranges::iterator_t;
+
+    {
+        int output[]    = {13, 42, 1367};
+        const int value = 7;
+        auto result     = fill(ranges::begin(output), ranges::end(output), value);
+        for (auto elem : output) {

```

Comment:
Style: I would prefer to see `for (const auto& elem : output)` which makes it clear that the iteration is in-place and non-modifying. (Although it doesn't make a difference here, it does in general; `for (auto elem : output)` initializes `auto elem = *iter;` which creates a distinct copy of each element.). Occurs below.